pkg/cmd/certregenerationcontroller/cabundlesyncer.go

```diff
@@ -0,0 +1,149 @@
+package certregenerationcontroller
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/operatorclient"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/targetconfigcontroller"
+)
+
+const workQueueKey = "key"
+
+type CABundleController struct {
+	kubeClient kubernetes.Interface
+
+	configMapLister corev1listers.ConfigMapLister
+
+	eventRecorder events.Recorder
+
+	cachesToSync []cache.InformerSynced
+
+	// queue only ever has one item, but it has nice error handling backoff/retry semantics
+	queue workqueue.RateLimitingInterface
+}
+
+func NewCABundleController(
+	kubeClient kubernetes.Interface,
+	kubeInformersForNamespaces v1helpers.KubeInformersForNamespaces,
+	eventRecorder events.Recorder,
+) (*CABundleController, error) {
+	c := &CABundleController{
+		kubeClient:      kubeClient,
+		configMapLister: kubeInformersForNamespaces.InformersFor(metav1.NamespaceAll).Core().V1().ConfigMaps().Lister(),
+		eventRecorder:   eventRecorder.WithComponentSuffix("manage-client-ca-bundle-recovery-controller"),
+		queue:           workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "CABundleRecoveryController"),
+	}
+
+	c.cachesToSync = append(c.cachesToSync, kubeInformersForNamespaces.InformersFor(metav1.NamespaceAll).Core().V1().ConfigMaps().Informer().HasSynced)
+
+	handler := cache.ResourceEventHandlerFuncs{
+		AddFunc:    func(obj interface{}) { c.queue.Add(workQueueKey) },
+		UpdateFunc: func(old, new interface{}) { c.queue.Add(workQueueKey) },
+		DeleteFunc: func(obj interface{}) { c.queue.Add(workQueueKey) },
+	}
+
+	// we react to some config changes
+	namespaces := []string{
```

Comment:
nit: i would make this global var with godoc on what this list means

Reply:
these namespaces are conventionally defined when setting the handlers https://github.com/openshift/cluster-kube-apiserver-operator/blob/2a44f96452f248818ff4c2467d51eef50fd139f2/pkg/operator/targetconfigcontroller/targetconfigcontroller.go#L86-L89

I'd prefer not to create a layer of indirection obfuscation which namespaces are watched.